### PR TITLE
Add support for 1-click login

### PR DESCRIPTION
**Pull request by am-on**
Ref: https://github.com/niteoweb/woocart-app/issues/180

### CODE REVIEW

File: tests/AutoLoginTest.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                'times'     => 2,
+            )
+        );
+        \WP_Mock::userFunction(
+            'wp_redirect', array(
+                'times' => 1,
+            )
+        );
+        $_GET['auth'] = 'foo_jwt_auth_token';
+        $_SERVER['SERVER_NAME'] = 'store.com';
```
Comment (am-on) [inline tests/AutoLoginTest.php]: tests stop if this line `$_SERVER['SERVER_NAME'] = 'store.com';`  is included

File: tests/AutoLoginTest.php [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                'return'  => [ true, false, ],
+                'times'   => 2,
+            )
+        );
+        \WP_Mock::userFunction(
+            'wp_redirect', array(
+                'times'   => 2,
+            )
+        );
+        \WP_Mock::userFunction(
```
Comment (am-on) [inline tests/AutoLoginTest.php]: tests stop if this is included

File: src/classes/class-autologin.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            if ( is_blog_installed() ) { // only run login functions on installed blog
+                add_action( 'init', array(&$this, 'test_for_auto_login') );
+                add_action( 'init', array(&$this, 'fix_login_auth'), ~PHP_INT_MAX );
+            }
+        }
+
+        /**
+         * Handle redirection to www without loosing auth parameter.
+         */
+        function fix_login_auth(): void {
```
Comment (dz0ny) [inline src/classes/class-autologin.php]: You can throw this out

File: src/classes/class-autologin.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	 * @package Niteo\WooCart\Defaults
+	 */
+	class AutoLogin {
+
+        /**
+         * AutoLogin constructor.
+         */
+        function __construct() {
+            if ( is_blog_installed() ) { // only run login functions on installed blog
+                add_action( 'init', array(&$this, 'test_for_auto_login') );
```
Comment (dz0ny) [inline src/classes/class-autologin.php]: hook to https://core.trac.wordpress.org/browser/tags/4.9.8/src/wp-login.php#L182

File: tests/AutoLoginTest.php (hunk 2 above)
Comment (am-on) [reply tests/AutoLoginTest.php]: removed `exit` from `test_for_auto_login`

File: src/classes/class-autologin.php (hunk 4 above)
Comment (am-on) [reply src/classes/class-autologin.php]: `login_header` is too late, headers are already sent
 ```
Cannot modify header information - headers already sent by
```

File: tests/fixtures/woocart_login_shared_secret [hunk 5]
```diff
@@ -0,0 +1 @@
+secret
```
Comment (dz0ny) [inline tests/fixtures/woocart_login_shared_secret]:
```suggestion
sharedSecret
```

You have it name like that in other PRs

**Pull request merged**